GVFS/GVFS.Common/ProductUpgrader.cs

```diff
@@ -141,16 +139,14 @@ public virtual bool TrySetupToolsDirectory(out string upgraderToolPath, out stri
         {
             string rootDirectoryPath = ProductUpgraderInfo.GetUpgradesDirectoryPath();
             string toolsDirectoryPath = Path.Combine(rootDirectoryPath, ToolsDirectory);
-            Exception exception;
-            if (!this.fileSystem.TryCreateDirectory(toolsDirectoryPath, out exception))
-            {
-                upgraderToolPath = null;
-                error = exception.Message;
-                this.TraceException(
-                    exception,
-                    nameof(this.TrySetupToolsDirectory),
-                    $"Error creating upgrade tools directory {toolsDirectoryPath}.");
-                return false;
+
+            if (!GVFSPlatform.Instance.FileSystem.TryCreateDirectoryWithAdminOnlyModify(

```

Comment:
On first read through - I expected this functionality to be provided by the class `fileSystem` instance. I think it would be preferable if this class only had 1 file system abstraction to deal with, if possible.

Reply:
> What is the distinction between `PhysicalFileSystem` and `IPlatformFileSystem`, and why did we chose to put the directory creation code in the latter.

The (not always perfect) distinction is that `IPlatformFileSystem` has any file system methods that need to be platform specific.  Typically this is because .NET does not provide a managed API for what we need to do, and so we need to PInvoke to the native API.

I do see now that in some other spots we moved the call to `GVFSPlatform.Instance.FileSystem` into `PhysicalFileSystem` and I think we could do the same here.